Competition/src/main/cpp/Drivetrain.cpp

```diff
@@ -7,15 +7,231 @@
 
 #include "Drivetrain.h"
 
-Drivetrain::Drivetrain() {}
-
-Drivetrain& Drivetrain::GetInstance()
+Drivetrain::Drivetrain() : boostMultiplier{0.5},
+                           kDriveKinematics{DriveConstants::kTrackwidth},
+                           kSimpleMotorFeedforward{RamseteConstants::kS, RamseteConstants::kV, RamseteConstants::kA},
+                           kTrajectoryConfigF{RamseteConstants::kMaxSpeed, RamseteConstants::kMaxAcceleration},
+                           kTrajectoryConfigR{RamseteConstants::kMaxSpeed, RamseteConstants::kMaxAcceleration},
+                           kDifferentialDriveVoltageConstraint{kSimpleMotorFeedforward, kDriveKinematics, 10_V},
+                           m_leftDriveLead{DriveConstants::CAN_ID_LEFT_LEAD, rev::CANSparkMax::MotorType::kBrushless},
+                           m_leftDriveFollowA{DriveConstants::CAN_ID_LEFT_FOLLOW_A, rev::CANSparkMax::MotorType::kBrushless},
+                           m_leftDriveFollowB{DriveConstants::CAN_ID_LEFT_FOLLOW_B, rev::CANSparkMax::MotorType::kBrushless},
+                           m_rightDriveLead{DriveConstants::CAN_ID_RIGHT_LEAD, rev::CANSparkMax::MotorType::kBrushless},
+                           m_rightDriveFollowA{DriveConstants::CAN_ID_RIGHT_FOLLOW_A, rev::CANSparkMax::MotorType::kBrushless},
+                           m_rightDriveFollowB{DriveConstants::CAN_ID_RIGHT_FOLLOW_B, rev::CANSparkMax::MotorType::kBrushless},
+                           m_odometry{frc::Rotation2d(units::degree_t(GetHeading()))},
+                           limelight_state{1} // Init as on so it immediately will turn off in the state machine
 {
-  // Guaranteed to be destroyed. Instantiated on first use.
+  kTrajectoryConfigF.SetKinematics(kDriveKinematics);
+  kTrajectoryConfigF.AddConstraint(kDifferentialDriveVoltageConstraint);
+  kTrajectoryConfigF.SetReversed(false);
+
+  kTrajectoryConfigR.SetKinematics(kDriveKinematics);
+  kTrajectoryConfigR.AddConstraint(kDifferentialDriveVoltageConstraint);
+  kTrajectoryConfigR.SetReversed(true);
+
+  imu.Calibrate();
+
+  InitDrives(rev::CANSparkMax::IdleMode::kCoast);
+}
+Drivetrain& Drivetrain::GetInstance() {
   static Drivetrain instance;
   return instance;
 }
 
 void Drivetrain::Periodic() {
-  // Update odometry
```

Comment:
Might we want to comment out what the sections do?

Reply:
Yes, I can add some documentation